### PR TITLE
Fix System.Runtime.Numerics Tests to run against desktop

### LINKED ISSUE #14991: Test failure: System.Runtime.Numerics.Tests failed with Xunit.Sdk.EqualException & Xunit.Sdk.TrueException
Opened on behalf of @jiangzeng

The test `System.Numerics.Tests.ComparisonTest/IComparable_Invalid` has failed.

Assert.Equal() Failure\r
    Expected: obj\r
    Actual:   (null)

        Stack Trace:

           at System.Numerics.Tests.ComparisonTest.IComparable_Invalid() in D:\A\_work\64\s\corefx\src\System.Runtime.Numerics\tests\BigInteger\Comparison.cs:line 367

System.Numerics.Tests.ComparisonTest/IComparable_Invalid:
```
Unhandled Exception of Type Xunit.Sdk.TrueException
Message :
Failure at line 299. Expected real: 1.5707963267949. Actual real: NaN\r\nExpected: True\r\nActual:   False

Stack Trace :
   at System.Numerics.Tests.ComplexTests.VerifyRealImaginaryProperties(Complex complex, Double real, Double imaginary, Int32 lineNumber) in D:\A\_work\64\s\corefx\src\System.Runtime.Numerics\tests\ComplexTests.cs:line 1583
   at System.Numerics.Tests.ComplexTests.ACos_Advanced(Double real, Double imaginary, Double expectedReal, Double expectedImaginary) in D:\A\_work\64\s\corefx\src\System.Runtime.Numerics\tests\ComplexTests.cs:line 299

```
Build : Master - 20170108.02 (Full Framework Tests)
Failing configurations:
- Windows.10.Amd64
  - AnyCPU-Debug
  - AnyCPU-Release

Failed tests:
System.Numerics.Tests.ComparisonTest/IComparable_Invalid
System.Numerics.Tests.ComplexTests/ACos_Advanced
System.Numerics.Tests.ComplexTests/Equals
System.Numerics.Tests.cast_fromTest/RunDoubleExplicitCastFromLargeBigIntegerTests
System.Numerics.Tests.logTest/RunLargeValueLogTests

Details:
https://mc.dot.net/#/product/netcore/master/source/official~2Fcorefx~2Fmaster~2F/type/test~2Ffunctional~2Fdesktop~2Fcli~2F/build/20170108.02/workItem/System.Runtime.Numerics.Tests

### LINKED ISSUE #14992: Test failure: System.Numerics.Tests.modpowTest/ModPow1Large2SmallInt_Threshold
Opened on behalf of @jiangzeng

The test `System.Numerics.Tests.modpowTest/ModPow1Large2SmallInt_Threshold` has failed.

System.TypeInitializationException : The type initializer for 'BigIntTools.Utils' threw an exception.\r
    ---- System.ArgumentNullException : Value cannot be null.\r
    Parameter name: type

        Stack Trace:

           at BigIntTools.Utils.RunWithFakeThreshold(String name, Int32 value, Action action)
       at System.Numerics.Tests.modpowTest.ModPow1Large2SmallInt_Threshold() in D:\A\_work\64\s\corefx\src\System.Runtime.Numerics\tests\BigInteger\modpow.cs:line 146
    ----- Inner Stack Trace -----
       at System.Reflection.IntrospectionExtensions.GetTypeInfo(Type type)
       at BigIntTools.Utils..cctor() in D:\A\_work\64\s\corefx\src\System.Runtime.Numerics\tests\BigInteger\BigIntTools.cs:line 38
Build : Master - 20170108.02 (Full Framework Tests)
Failing configurations:
- Windows.10.Amd64
  - AnyCPU-Debug
  - AnyCPU-Release

Failed tests:
System.Numerics.Tests.IsEvenTest/RunIsEvenTests
System.Numerics.Tests.modpowTest/ModPow1Large2SmallInt_Threshold
System.Numerics.Tests.modpowTest/ModPow2Large1SmallInt_Threshold
System.Numerics.Tests.modpowTest/ModPow3LargeInt
System.Numerics.Tests.multiplyTest/RunMultiply_TwoLargeBigIntegers_Threshold

Details:
https://mc.dot.net/#/product/netcore/master/source/official~2Fcorefx~2Fmaster~2F/type/test~2Ffunctional~2Fdesktop~2Fcli~2F/build/20170108.02/workItem/System.Runtime.Numerics.Tests

### LINKED ISSUE #14993: Test failure: System.Numerics.Tests.powTest/RunOverflow
Opened on behalf of @jiangzeng

The test `System.Numerics.Tests.powTest/RunOverflow` has failed.

Assert.Throws() Failure\r
    Expected: typeof(System.OverflowException)\r
    Actual:   typeof(System.OutOfMemoryException): Array dimensions exceeded supported range.

        Stack Trace:

           at System.Numerics.BigIntegerBuilder.SetSizeClear(Int32 cu)
       at System.Numerics.BigIntegerBuilder.Mul(BigIntegerBuilder& reg1, BigIntegerBuilder& reg2)
       at System.Numerics.BigInteger.Pow(BigInteger value, Int32 exponent)
       at System.Numerics.Tests.powTest.<>c__DisplayClass9_0.<RunOverflow>b__0() in D:\A\_work\64\s\corefx\src\System.Runtime.Numerics\tests\BigInteger\pow.cs:line 178
Build : Master - 20170108.02 (Full Framework Tests)
Failing configurations:
- Windows.10.Amd64
  - AnyCPU-Debug
  - AnyCPU-Release

Details:
https://mc.dot.net/#/product/netcore/master/source/official~2Fcorefx~2Fmaster~2F/type/test~2Ffunctional~2Fdesktop~2Fcli~2F/build/20170108.02/workItem/System.Runtime.Numerics.Tests/analysis/xunit/System.Numerics.Tests.powTest~2FRunOverflow

### PR DESCRIPTION
**Pull request by tarekgh**
Fixes #14993
Fixes  #14992
Fixes  #14991

### CODE REVIEW

**Comment by tarekgh** [comment]
@mellinoe could you please have a look?

cc @danmosemsft

File: src/System.Runtime.Numerics/tests/BigInteger/Comparison.cs [hunk 1]
```diff
@@ -359,12 +359,25 @@ private static void RunNegativeTests(Random random)
...
         }
 
-        [Fact]
-        public static void IComparable_Invalid()
+        public static void IComparable_Invalid(string paramName)
         {
             IComparable comparable = new BigInteger();
             Assert.Equal(1, comparable.CompareTo(null));
-            Assert.Throws<ArgumentException>("obj", () => comparable.CompareTo(0)); // Obj is not a BigInteger
+            Assert.Throws<ArgumentException>(paramName, () => comparable.CompareTo(0)); // Obj is not a BigInteger
```
Comment (mellinoe) [inline src/System.Runtime.Numerics/tests/BigInteger/Comparison.cs]: Do we want/need these exception messages to be the same?

File: src/System.Runtime.Numerics/tests/BigInteger/IsEven.cs [hunk 2]
```diff
@@ -13,6 +13,7 @@ public class IsEvenTest
         private static int s_seed = 0;
 
         [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
```
Comment (mellinoe) [inline src/System.Runtime.Numerics/tests/BigInteger/IsEven.cs]: I think we actually need to figure out why the behavior is different here. This one doesn't seem to be doing anything with internal state.

File: src/System.Runtime.Numerics/tests/ComplexTests.cs [hunk 3]
```diff
@@ -292,6 +292,7 @@ public static IEnumerable<object[]> ACos_Advanced_TestData()
         }
 
         [Theory, MemberData("ACos_Advanced_TestData")]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
```
Comment (mellinoe) [inline src/System.Runtime.Numerics/tests/ComplexTests.cs]: I think all of the tests in this class should be checked as well; they don't seem to be testing anything platform-specific.

File: src/System.Runtime.Numerics/tests/BigInteger/Comparison.cs (hunk 1 above)
Comment (tarekgh) [inline src/System.Runtime.Numerics/tests/BigInteger/Comparison.cs]: It doesn't have to be. it depends on the behavior we desire. if netcore behavior is correct or better one then we can have differences. if it doesn't matter to have this netcore behavior then we can change it to match the desktop. from what I am seeing it make sense to keep netcore behavior

**Comment by tarekgh** [comment]
> Most of the changes look good; we should disable the tests that are netcoreapp-specific (the ones that are using internal state). Some of the tests look pretty normal, though. I think we need to take a closer look to figure out why they are behaving differently. Are we able to mark those tests with ActiveIssue for .NET Framework instead?

I am just trying to enable the test to run on the desktop. I have opened the issue #15051 to track this so we  don't have to block this PR and get the test runs on desktop. please let me know if you are ok with this.

**Comment by mellinoe** [comment]
I'm okay with that. It sounds like we can't use ActiveIssue for a particular TFM, so this is fine for now.

**Comment by tarekgh** [comment]
test Innerloop Ubuntu14.04 Debug Build and Test please

**Comment by tarekgh** [comment]
test Innerloop OSX Debug Build and Test please

File: src/System.Runtime.Numerics/tests/BigInteger/Comparison.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Runtime.Numerics/tests/BigInteger/Comparison.cs]: Another option is just to remove the name verification in this case.  What you have is fine, though.

**Pull request merged**